functions_and_scope_assingnment.js

```diff
@@ -0,0 +1,329 @@
+}
+getDogAge(3);
+//
+//
+//
+// ## Question 3
+//
+// Write a function `arraySum` that takes in an array and returns the sum of all the elements.
+//
+// Example:
+//
+// Input: `[13,2,4,-2]`
+//
+// Output: `17`
+let newArray = [13,2,4, -2]
+
+function arraySum(array){
+  let sum = 0;
+  for(i = 0; i < array.length; i ++){
+    sum += array[i];
+  }
+  console.log(sum);
+
+}
+
+
+arraySum(newArray);
+//
+//
+//
+//
+// ## Question 4
+//
+// Write a function that takes in any two numbers and return their sum.
+//
+// Example:
+//
+// Input: `14, 5`
+//
+// Output: `19`
+function sumFinder(sum1, sum2){
+  let total = sum1 + sum2;
+  return total;
+}
+
+console.log(sumFinder(3,4));
+
+
+// ## Question 5
+//
+// Write a function takes in any number grade and returns a corresponding letter grade.
+//
+// | Number Grade | Equivalent Letter Grade |
+// | :--------: | :---------: |
+// | 100 | A+ |
+// | 90 - 99 | A |
+// | 80 - 89 | B |
+// | 70 - 79 | C |
+// | 65 - 69 | D |
+// | Below 65 | F |
+//
+//
+// Example:
+//
+// Input: `88`
+//
+// Output: `"B"`
+//
+function gradeCalculator(grade){
+  if (grade >=100){
+    console.log("You recieved and A+");
+  }else if (grade >= 90 && grade <= 99){
+    console.log("You recieved and A");
+  }else if (grade >= 80 && grade <= 89){
+    console.log("You recieved and B");
+  }else if (grade >=70 && grade <= 79){
+    console.log("You recieved and C");
+  }else if (grade >= 65 && grade <= 69){
+    console.log("You recieved and D");
+  }else if (grade <= 65){
+    console.log("You recieved and F");
+  }else{
+    console.log("Please put in a number value");
+  }
+}
+
+gradeCalculator("2");
+
+
+
+// ## Question 6
+//
+// Implement a function named `repeatLog` that takes a string and a number as parameters.
+//The function should log `message` `count` number of times.
+//
+// Example:
+//
+// Input: `"+", 10`
+//
+// Output: `++++++++++`
+function repeatLog(str, repeatX){
+  for ( x = 1; x <= repeatX; x ++){
+    console.log(`${str} ${x} number of ${repeatX} times`);
+  }
+}
+
+repeatLog("This is message", 10);
+
+
+
+
+// ## Question 7
+//
+// Write a function named `first` that takes a number and returns an array with all the numbers from 1 to n
+//
+//
+// Example:
+//
+// Input: `first(3)`
+//
+// Output: `[1, 2, 3]`
+function first(num){
+  let up = [];
+  for (y = 1; y <= num; y ++){
+    up.push(y);
+  }
+  console.log(up) ;
+}
+first(3)
+
+
+
+// ## Question 8
+//
+// Write a function that logs the numbers from 1 to x, except:
+//
+// If the number if a multiple of 3, log `"Fizz"` instead of the number
+// If the number is a multiple of 5, log `"Buzz"` instead of the number
+// If the number is a multiple of 3 AND 5, log `"FizzBuzz"` instead of the number
+// Your function should take in one parameter: x (the number to count up to)
+//
+function fizzBuzz(fizz){
+let z = 0;
+  while (z <= fizz){
+    //log numbers % 3
+    if (z % 3 === 0 && z % 5 !== 0){// divisible by 3 and not 5{
```

Comment:
Although accurate, you could simple all three if conditions by saying modulo 3, modulo 5 and modulo 15. Your logic still works though.